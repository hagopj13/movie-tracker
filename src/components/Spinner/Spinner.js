import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    margin: theme.spacing(3, 0),
  },
}));

const Spinner = () => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <CircularProgress color="primary" />
    </div>
  );
};

export default Spinner;
