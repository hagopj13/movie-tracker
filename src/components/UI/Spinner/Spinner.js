import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    margin: theme.spacing(3, 0),
  },
}));

export default function CircularIndeterminate() {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <CircularProgress color="primary" />
    </div>
  );
}
