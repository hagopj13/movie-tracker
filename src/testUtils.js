import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { createMount } from '@material-ui/core/test-utils';

import { initialState } from 'store/rootReducer';

const mount = createMount();
const mockStore = configureMockStore()(initialState);

export const mountWithStore = (children, store = mockStore) =>
  mount(<Provider store={store}>{children}</Provider>);
