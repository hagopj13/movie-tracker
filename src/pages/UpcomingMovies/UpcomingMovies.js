// @flow
import React, { useLayoutEffect } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';

import Spinner from 'components/Spinner/Spinner';
import upcomingActions from 'store/upcoming/upcoming.actions';
import loadingSelectors from 'store/api/loading/loading.selectors';
import ConfigActionTypes from 'store/config/config.types';

import UpcomingMoviesList from './List/UpcomingMoviesList';

type Props = {
  isLoadingConfig: boolean,
  onFetchMovies: () => void,
};

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    justifyContent: 'center',
  },
  container: {
    padding: theme.spacing(5, 5, 2),
    display: 'flex',
  },
  movieListContainer: {
    flexGrow: 1,
  },
}));

export const UpcomingMovies = (props: Props) => {
  const { isLoadingConfig, onFetchMovies } = props;

  const classes = useStyles();

  useLayoutEffect(() => {
    onFetchMovies();
  }, [onFetchMovies]);

  if (isLoadingConfig) {
    return <Spinner />;
  }

  return (
    <div className={classes.root}>
      <Container className={classes.container}>
        <div className={classes.movieListContainer}>
          <UpcomingMoviesList />
        </div>
      </Container>
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  isLoadingConfig: loadingSelectors.createIsLoadingSelector([
    ConfigActionTypes.FETCH_IMAGES_CONFIG,
  ]),
});

const mapDispatchToProps = {
  onFetchMovies: upcomingActions.fetchMovies,
};

export default connect(mapStateToProps, mapDispatchToProps)(UpcomingMovies);
