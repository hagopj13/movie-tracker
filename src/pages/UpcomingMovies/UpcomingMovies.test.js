import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import { mountWithStore } from 'testUtils';
import Spinner from 'components/Spinner/Spinner';

import { UpcomingMovies } from './UpcomingMovies';

describe('UpcomingMovies page', () => {
  let shallow;
  let wrapper;

  const mockFetchMovies = jest.fn();

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render the UpcomingMovies page correctly', () => {
    const mockProps = {
      isLoadingConfig: false,
      onFetchMovies: mockFetchMovies,
    };
    wrapper = shallow(<UpcomingMovies {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a Spinner when config is loading', () => {
    const mockProps = {
      isLoadingConfig: true,
      onFetchMovies: mockFetchMovies,
    };
    wrapper = shallow(<UpcomingMovies {...mockProps} />);
    expect(wrapper.exists(Spinner)).toBe(true);
  });

  it('should fetch movies after mounting', () => {
    const mockProps = {
      isLoadingConfig: false,
      onFetchMovies: mockFetchMovies,
    };
    wrapper = mountWithStore(<UpcomingMovies {...mockProps} />);
    expect(mockFetchMovies).toHaveBeenCalledTimes(1);
  });
});
