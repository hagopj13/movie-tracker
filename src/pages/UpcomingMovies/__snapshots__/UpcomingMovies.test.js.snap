// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpcomingMovies page should render the UpcomingMovies page correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-2"
  >
    <div
      className="makeStyles-movieListContainer-3"
    >
      <Connect(UpcomingMoviesList) />
    </div>
  </WithStyles(ForwardRef(Container))>
</div>
`;
